handlers/pulp_rpm/handlers/rpmtools_zypper.py

```diff
@@ -0,0 +1,563 @@
+        """
+        Install packages by name.
+        :param names: A list of package names.
+        :type names: [str,]
+        :return: Packages installed.
+            {resolved=[Package,],deps=[Package,], failed=[Package,]}
+        :rtype: dict
+        """
+
+        result = None
+        names = [str(x.replace('-*-*.*', '').replace('*:', '')) for x in names]
+
+        try:
+            log.debug("Starting process for installing %s", names)
+
+            q = Queue()
+            p = Process(target=self.install_p, args=(names, q, ))
+            p.start()
+
+            p.join()
+            result = q.get()
+        except Exception:
+            var = traceback.format_exc()
+            log.error("Exception happend during install: %s", var)
+        finally:
+            log.debug("Install transaction finished.")
+
+        return result
+
+    def uninstall_p(self, names, q):
+        z = Zypp()
+        for package in names:
+            log.info('Want to uninstall package %s', package)
+            z.poolRemove(package)
+
+        if z.poolResolve():
+            result = z.tx_summary("uninstall")
+            z.poolCommit(USE_DRY_RUN)
+        z.finish()
+        q.put(result)
+
+    def uninstall(self, names):
+        """
+        Uninstall (erase) packages by name.
+        :param names: A list of package names to be removed.
+        :type names: list
+        :return: Packages uninstalled (erased).
+            {resolved=[Package,],deps=[Package,], failed=[Package,]}
+        :rtype: dict
+        """
+
+        result = None
+
+        # TODO: looks like that this is not necessary for "uninstall". Only str() is required!
+        names = [str(x.replace('-*-*.*', '').replace('*:', '')) for x in names]
+
+        try:
+            log.debug("Starting process for uninstalling %s", names)
+
+            q = Queue()
+            p = Process(target=self.uninstall_p, args=(names, q, ))
+            p.start()
+
+            p.join()
+            result = q.get()
+        except Exception:
+            var = traceback.format_exc()
+            log.error("Exception happend during uninstall: %s", var)
+        finally:
+            log.debug("Uninstall transaction finished.")
+
+        return result
+
+    def update_p(self, names, q):
+        z = Zypp()
+        if names:
+            for package in names:
+                log.info('Want to update package %s', package)
+                z.poolUpdate(package)
+
+                if z.poolResolve():
+                    result = z.tx_summary("update")
+                    z.poolCommit(USE_DRY_RUN)
+        else:
+            log.info('Want to update all packages')
+            z.poolUpdateAll()
+            result = z.tx_summary("update")
+            z.poolCommit(USE_DRY_RUN)
+        z.finish()
+        q.put(result)
+
+    def update(self, names=()):
+        """
+        Update installed packages.
+        When (names) is not specified, all packages are updated.
```

Comment:
Would it be ok to separate the parameters from the docstring body with one line of whitespace. Here and similar elsewhere?

Reply:
It would be OK but the rpmtools_yum.py doesn't do that. Therefore I think it's better be similar to rpmtools_yum.py and let it as it is.